src/popup/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { dimensions } from '@dojo/framework/core/middleware/dimensions';
+import { theme } from '@dojo/framework/core/middleware/theme';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import * as css from '../theme/popup.m.css';
+import * as fixedCss from './popup.m.css';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+
+export type PopupPosition = 'above' | 'below';
+
+export interface PopupProperties {
+	/** Property to define if the popup wrapper should match the trigger width, defaults to true */
+	matchWidth?: boolean;
+	/** Property to define where the popup should render relative to the trigger, defaults to below */
+	position?: PopupPosition;
+	/** Property to define if the underlay should be visible, defaults to false */
+	underlayVisible?: boolean;
+}
+
+interface PopupICache {
+	open: boolean;
+}
+
+export interface PopupChildren {
+	trigger: (onToggleOpen: () => void) => RenderResult;
+	content: (onClose: () => void) => RenderResult;
+}
+
+const icache = createICacheMiddleware<PopupICache>();
+
+const factory = create({ dimensions, theme, icache })
+	.properties<PopupProperties>()
+	.children<PopupChildren>();
+
+export const Popup = factory(function({
+	properties,
+	children,
+	middleware: { dimensions, theme, icache }
+}) {
+	const { underlayVisible = false, position = 'below', matchWidth = true } = properties();
+
+	const wrapperDimensions = dimensions.get('wrapper');
+	const { position: triggerPosition, size: triggerSize } = dimensions.get('trigger');
+	const triggerTop = triggerPosition.top + document.documentElement.scrollTop;
+	const triggerBottom = triggerTop + triggerSize.height;
+	const bottomOfVisibleScreen =
+		document.documentElement.scrollTop + document.documentElement.clientHeight;
+	const topOfVisibleScreen = document.documentElement.scrollTop;
+
+	const willFit = {
+		below: triggerBottom + wrapperDimensions.size.height <= bottomOfVisibleScreen,
+		above: triggerTop - wrapperDimensions.size.height >= topOfVisibleScreen
+	};
+
+	let wrapperStyles: Partial<CSSStyleDeclaration> = {
+		opacity: '0'
+	};
+
+	if (wrapperDimensions.size.height) {
+		wrapperStyles = {
+			width: matchWidth ? `${triggerSize.width}px` : 'auto',
+			left: `${triggerPosition.left}px`,
+			opacity: '1'
+		};
+
+		if (position === 'below') {
+			if (willFit.below) {
+				wrapperStyles.top = `${triggerBottom}px`;
+			} else {
+				wrapperStyles.top = `${triggerTop - wrapperDimensions.size.height}px`;
```

Comment:
Do we need to handle when it won't fit above or below? I'm thinking this could happen with a mobile phone on landscape.

Reply:
I plan for us to tackle left / right positions as a follow up PR / issue

Reply:
Should we get an issue posted just so we don't lose track of this?

Reply:
I added it to the existing issue already @vansimke